mm2src/common/executor/native_executor.rs

```diff
@@ -6,12 +6,14 @@ use std::pin::Pin;
 use std::thread;
 use std::time::Duration;
 
-pub fn spawn(future: impl Future03<Output = ()> + Send + 'static) { crate::wio::CORE.0.spawn(future); }
-
-pub fn spawn_boxed(future: Box<dyn Future03<Output = ()> + Send + Unpin + 'static>) { spawn(future); }
+/// # Safety
+///
+/// The `spawn` function must be used carefully to avoid hanging pointers.
+/// Please consider using `AbortableQueue`, `AbortableSimpleMap` or `spawn_abortable` instead.
+pub unsafe fn spawn(future: impl Future03<Output = ()> + Send + 'static) { crate::wio::CORE.0.spawn(future); }
```

Comment:
As per https://doc.rust-lang.org/book/ch19-01-unsafe-rust.html#unsafe-superpowers, unsafe code has 5 "superpowers", none of which are actually used here. It doesn't seem right to make some fn unsafe when it doesn't match the actual definition. E.g., in case of memory-related issues, other devs might start wrongfully look for them in `spawn`.

Reply:
Making this function safe and public may lead to someone to use it without checking the documentation. On the other hand, `unsafe` keyword causes some kind of danger (at least for me), and it can be the stop signal.

Also `unsafe` keyword doesn't affect the compiler checks in any way:
> unsafe doesn’t turn off the borrow checker or disable any other of Rust’s safety checks

Reply:
It's not about compiler checks. It's about using something that semantically has a very different meaning. `unsafe` is not intended to be used for "you should check the docs" purpose :)

As an alternative: is there a way to make this fn private instead?

Reply:
Unfortunately, no, but we can rename it to `__spawn`, or do something like
```rust
pub mod private {
  pub fn spawn();
}
```
Then this function will be used as `private::spawn()`. What do you think?

Reply:
`pub mod private` is inconsistent and looks even more confusing to me than `unsafe`. I will try to figure out a possible refactoring to make spawn private and push a demo commit if I'm able to :)

Reply:
So, with this diff (a bit dirty) https://github.com/KomodoPlatform/atomicDEX-API/compare/stop-spawned-futures...KomodoPlatform:atomicDEX-API:spawn-pub-crate?expand=1, I was able to make `spawn` at least `pub(crate)`, which could be sufficient to resolve this comment. The only problem is `stop` RPC - without delay and spawn I have some tests failing with `Transport 'http://127.0.0.197:7783/' error: connection closed before message completed`, and it's likely to happen too when it's used by GUIs with unexpected outcome. @sergeyboyko0791 Do you maybe have an idea how to work this around? 🙂